rpcs3/Crypto/ec.cpp

```diff
@@ -103,17 +97,14 @@ static constexpr u8 inv256[0x80] = {
 	0x11, 0x3b, 0x5d, 0xc7, 0x49, 0x33, 0x55, 0xff,
 };
 
-static void bn_mon_muladd_dig(u8* d, u8* a, u8 b, u8* N, u32 n)
+static void bn_mon_muladd_dig(u8* d, const u8* a, u8 b, const u8* N, u32 n)
 {
-	u32 dig;
-	u32 i;
-
 	const u8 z = -(d[n - 1] + a[n - 1] * b) * inv256[N[n - 1] / 2];
 
-	dig = d[n - 1] + a[n - 1] * b + N[n - 1] * z;
+	u32 dig = d[n - 1] + a[n - 1] * b + N[n - 1] * z;
 	dig >>= 8;
 
-	for (i = n - 2; i < n; i--)
+	for (u32 i = n; i >= 2; i--)
```

Comment:
Is it actually the same? i is used for indexing.

Reply:
It looks like it's relying on the overflow to exit the loop, which is an ill defined loop.